src/common/coding/test/roundTrip.test.js

```diff
@@ -264,6 +264,29 @@ describe('graph', () => {
     // const decoded = decodeGraph(encoded);
     // console.log(decoded);
   });
+
+  test('nested_empty_object', () => {
+    roundTrip(
+      {
+        person: [
+          {
+            $key: { example: 'filter', $cursor: ['something'] },
+            $ref: ['person', 'exampleId'],
+            domains: {},
+          },
+        ],
+      },
+      {
+        person: [
+          {
+            $key: { example: 'filter', $cursor: ['something'] },
+          },
+        ],
+      },
```

Comment:
Is this 2nd element of this array required? It wasn't in the original test case. What is the use of this if I may ask?

Reply:
It's not a second element in the array, it's a second argument to the roundTrip function.

Roundtrip (defined above) is called with (original, expected, extraAssertions). We're passing expected, because here the expected value after encode & decode is slightly different from original (basically, $ref becomes a non-enumerable property).